packages/protocol/contracts/identity/Attestations.sol

```diff
@@ -572,6 +572,11 @@ contract Attestations is
       seed = keccak256(abi.encodePacked(seed));
       validator = validatorSignerAddressFromCurrentSet(uint256(seed) % numberValidators);
       issuer = getAccounts().validatorSignerToAccount(validator);
+
+      if (!getAccounts().hasAuthorizedAttestationSigner(issuer)) {
```

Comment:
There's an optimization we can make here, which is to, outside of the while loop, set `IAccounts accounts = getAccounts();`, and then inside the loop call `accounts.validatorSignerToAccount()/accounts.hasAuthorizedAttestationSigner()`